### PR TITLE
Shotgun Arm CBM

### PR DESCRIPTION
**Pull request by nexusmrsep**
#### Summary
 ```SUMMARY: Features "Shotgun Arm CBM - weapon concealed in your arm"```

#### Purpose of change
Idea from Discourse Forum.
Basically a sawn-off one barrel one-shot shotgun concealed in an arm. Activate it, your arm transforms into a shotgun, fire.
Seen on Ghost in The Shell.

#### Describe the solution
Added new CBM weapon. From the code side of view, for it to work properly its a strange chimera, its technically a "melee weapon cbm" (not a "gun cbm", since bio-guns use power instead of ammo) and also its "reload&fire" type - which was needed for it to properly use ammo - so it behaves a bit like a bow. However the item itself is a proper gun. This was needed because there is no other way for it to remember if it had ammo loaded after retracting it back, so it pops-up loaded again.

#### Describe alternatives you've considered
Re-balancing it's properties around other numbers, but it's mostly based on base shotgun with dispersion of a a hand made pipe-shotgun. Perhaps not the best of weapons, but its basically a barrel in an arm. It get the job done as a backup weapon in a pinch.

#### Additional context
N/A

### CODE REVIEW

**Comment by SourSlime** [comment]
Sounds like the kind of thing that should damage the player from recoil if they don’t have strong joints/bones/converted their other arm to a Fusion Blaster (that way they can’t hold the gun steady with their other arm)

**Comment by thethunderhawk** [comment]
>Sounds like the kind of thing that should damage the player from recoil [under certain circumstances].

Idk about that. From a lore standpoint I think that’s easy to work around (internal shock absorbers or whatever), and from a gameplay standpoint it’s probably already going to be a low-tier CBM (being basically just a sawn off shotgun you don’t have to use volume or encumbrance to carry) and it doesn’t need to be weakened further.

**Comment by Fracture64** [comment]
Movie Upgrade featured some cool internal weapon implants: https://www.youtube.com/watch?v=36PDeN9NRZ0&feature=youtu.be&t=103

From what I remember that technology was restricted to military, and functionally was a semi auto shotgun (there also was pistol version but it got a lot less screen time). Loading gate was above elbow though, which doesn't make sense. "Realistically" it would be located below so implant is only located in forearm. We could figure out maximum amount of shells by taking average human forearm length and dividing by shotgun shell (which one by the way? Probably should be restricted to smallest gauge in game).

**Comment by nexusmrsep** [comment]
That reminds me that I totally forgot to add it to the distribution tables.

**Comment by Fracture64** [comment]
Also could be a fitting addition to Bionic Assassin, doesn't seem to fit any others professions. Razor Boy seems to be more focused on melee and Bionic Thief is not combat oriented.

**Comment by nexusmrsep** [comment]
Another though: it probably is a good time to think of CBM cancel-out lists or bionic slots. I don't want this to stack with fusion blaster arm, but I also don't want to hard-code the mutual exclusions.

**Comment by Fris0uman** [comment]
What would make limb replacement CBM really neat would be the ability to lose limbs. Lost my arm ? Ho well let's put a shotgun there.

**Comment by SolsticeTheTechMarine** [comment]
Man, this sounds a lot like Trigun to me.

**Comment by nexusmrsep** [comment]
This doesn't replace an arm fortunately, but limb prosthetics is something I'd like to do, if it wasn't a hard nut to crack. Consider a bionic arm that is armoured by definition, that would need to be repair if it loses HP instead of natural healing of an organic limb. If it was possible code wise, then it would be possible to simulate limb loss without condemning player to certain death. This however is hard to achieve as it would require some serious changes in anatomy, healing process and other interactions. But never say never.

**Comment by dissociativity** [comment]
If it doesn't replace arm, make it lower arm HP by 25%

**Comment by kevingranade** [comment]
Alternately to lowering HP, it could add arm encumbrance since it's heavier than the none it replaces.

**Comment by paulenka-aleh** [comment]
Oh, finally a right way to surprise NPCs asking me to drop my freaking whatever to rob me afterwards

**Comment by nexusmrsep** [comment]
I've added +5 encumbrance.
Should I add left-arm/right-arm variants or keep it always-left-arm only?

**Comment by nexusmrsep** [comment]
I'm going to upgrade bionics code to allow bionic shotgun (and potential future bionic weapons) to store ammo when hidden in body.

**Comment by nexusmrsep** [comment]
Done. Integrated shotgun can now store ammo within, while being retracted so it can be activated and deployed loaded. Also added code that prompts fire menu to fire the gun when its being activated, but only when its loaded.

File: src/bionics.cpp [hunk 1]
```diff
@@ -1591,6 +1600,8 @@ void bionic::deserialize( JsonIn &jsin )
     invlet = jo.get_int( "invlet" );
     powered = jo.get_bool( "powered" );
     charge = jo.get_int( "charge" );
+    ammo_loaded = jo.get_string( "ammo_loaded" );
```
Comment (ZhilkinSerg) [inline src/bionics.cpp]: Causes #29336. Should use some fallback values, i.e.:

```CPP
    ammo_loaded = jo.get_string( "ammo_loaded", "null" );
    ammo_loaded = jo.get_string( "ammo_count", 0 );
```

**Comment by kevingranade** [comment]
This pull request has been mentioned on **Cataclysm: Dark Days Ahead**. There might be relevant details there:

https://discourse.cataclysmdda.org/t/what-are-some-bionics-youve-always-wanted-to-see/19101/56

**Pull request merged**